service_mesh/v2x/ossm-extensions.adoc

```diff
@@ -0,0 +1,166 @@
+
+== WebAssembly Extensions
+
+WebAssembly modules can be run on many platforms, including proxies, and has broad language support, fast execution with near-native speed and a sandboxed-by-default security model.
+
+.Extension Capabilities
+
+{ProductName} Extensions are link:https://www.envoyproxy.io/docs/envoy/v1.16.0/intro/arch_overview/http/http_filters#arch-overview-http-filters[Envoy HTTP Filters], giving them a wide range of capabilities:
+
+* Manipulating the body and headers of requests and responses
+* Out-of-band HTTP requests to services not in the request path, such as authentication or policy checking
+* Side-channel data storage and queues for filters to communicate with each other
+
+There are two parts to writing a {ProductName} Extension: you'll have to write your extension using an SDK that exposes the link:https://github.com/proxy-wasm/spec[proxy-wasm API] and compile it to a WebAssembly module, and then package it into a container.
+
+.Supported languages
+
+You can use any language that compiles to WebAssembly bytecode to write a {ProductName} Extension, but the following languages have existing SDKs that expose the proxy-wasm API so that it can be consumed directly.
+
+.Supported languages
+|===
+| Language | Maintainer | Repository
+
+| C++ 
+| proxy-wasm team (Istio Community) 
+| link:https://github.com/proxy-wasm/proxy-wasm-cpp-sdk[proxy-wasm/proxy-wasm-cpp-sdk]
+
+| Rust 
+| proxy-wasm team (Istio Community) 
+| link:https://github.com/proxy-wasm/proxy-wasm-rust-sdk[proxy-wasm/proxy-wasm-rust-sdk]
+
+| AssemblyScript 
+| solo.io 
+| link:https://github.com/solo-io/proxy-runtime[solo-io/proxy-runtime]
+
+| Go 
+| tetrate.io 
+| link:https://github.com/tetratelabs/proxy-wasm-go-sdk[tetratelabs/proxy-wasm-go-sdk]
+|===
+
+=== Container Format
+
+You must have a `.wasm` file containing the bytecode of your WebAssembly module, and a `manifest.yaml` file in the root of the container filesystem to make your container image a valid extension image.
+
+.manifest.yaml
+[source,yaml]
+----
+schemaVersion: 1
+  
+name: <your-extension>
+description: <description>
+version: 1.0.0
+phase: PreAuthZ
+priority: 100
+module: extension.wasm
+----
+
+.manifest.yml Field Reference
+|===
+| Field | Description
+
+|schemaVersion
+|Used for versioning of the manifest schema. Currently the only possible value is `1`.
+
+|name
+|The name of your extension. This field is just metadata and currently unused.
+
+|description
+|The description of your extension. This field is just metadata and currently unused.
+
+|version
+|The version of your extension. This field is just metadata and currently unused.
+
+|phase
```

Comment:
Is there a list of valid values for Phase?

ETA - Found it later in the document.